### PR TITLE
Feature/main functionality

### PR DESCRIPTION
**Pull request by nedarokt**
MP-577: Update user agent to have RoktWebViewSDK
MP-579: Create the interface for capturing RoktWebViewSDK.open(url)
MP-369: Implement executing external default browser from Rokt webview SDK

### CODE REVIEW

File: roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/Constants.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ *  You may not use this file except in compliance with the License.
+ *
+ *  You may obtain a copy of the License at https://rokt.com/sdk-license-2-0/
+ */
+
+package rokt.com.roktwebviewsdk
+
+object Constants {
+    internal const val WEB_INTERFACE_NAME = "RoktWebViewSDK"
+    internal const val ROKT_USER_AGENT_STRING = " $WEB_INTERFACE_NAME/${BuildConfig.VERSION_NAME}"
```
Comment (danial-rokt) [inline roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/Constants.kt]: better to have 1.0 as the version name is 1.0.0

File: app/src/main/java/rokt/com/roktwebviewsample/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+package rokt.com.roktwebviewsample
+
+import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+    private val sampleHtmlString =
```
Comment (danial-rokt) [inline app/src/main/java/rokt/com/roktwebviewsample/MainActivity.kt:20]: do you think it is a good idea to have a text view(URL) and a button(like "go") that opens a URL and also have a mock button to open this string for testing?
in that case, we could support mocked and online testing on the same application

File: roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/RoktWebView.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import rokt.com.roktwebviewsdk.Constants.ROKT_USER_AGENT_STRING
+import rokt.com.roktwebviewsdk.Constants.WEB_INTERFACE_NAME
+import java.lang.ref.WeakReference
+
+/**
+ * To use, replace instances of WebView with RoktWebView.
+ * Calls to RoktWebViewSDK.open(url) will open in an external browser.
+ */
+@SuppressLint("SetJavaScriptEnabled")
+class RoktWebView @JvmOverloads constructor(
```
Comment (DJingSeo) [inline roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/RoktWebView.kt:23]: Would it be possible to have a few unittest against this class to confirm that it adds suffixes well

File: app/src/main/java/rokt/com/roktwebviewsample/MainActivity.kt (hunk 2 above)
Comment (nedarokt) [reply app/src/main/java/rokt/com/roktwebviewsample/MainActivity.kt:20]: Good idea, I'll add it

File: roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/Constants.kt (hunk 1 above)
Comment (nedarokt) [reply roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/Constants.kt]: Should we then change our version name to 1.0? I thought this [version] field needs to match the actual version name?

File: roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/RoktWebView.kt (hunk 3 above)
Comment (nedarokt) [reply roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/RoktWebView.kt:23]: Yep I'll add some UI tests

File: roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/Constants.kt (hunk 1 above)
Comment (danial-rokt) [reply roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/Constants.kt]: I just double-checked. We would only use [major.minor] of the semantic version but we would keep semantic versioning for the SDK. What do you think?

Comment (nedarokt) [reply roktwebviewsdk/src/main/java/rokt/com/roktwebviewsdk/Constants.kt]: Sounds good

**Review by danial-rokt — APPROVED** [review]

**Pull request merged**